exercise-app.js

```diff
@@ -28,13 +19,59 @@ const Weather = ({ temp, description, icon }) => (
 );
 
 
-const App = () => (
+class App extends React.Component {
+
+  constructor(props) {                // where do we hook up these props to ? where does this prop data come from?
+  super(props);                       // what is the purpose of this? why do we need to call super?
```

Comment:
You'll notice that `App` extends `React.Component`. `React.Component` class has all the lifeCycle ✨ magic such as `componentDidMount`, `componentWillUnmount`, etc. To make `React.Component` aware of our component we need to pass the props being passed into the `App` component into `React.Component`. 

When you extend a JS then you can talk to it using the `super` keyword.

`AppParentComponent` ➡️ _props_ ➡️ `App` ➡️ _`super(props)`_  ➡️  React.Component ➡️ ✨ lifeCycle ✨